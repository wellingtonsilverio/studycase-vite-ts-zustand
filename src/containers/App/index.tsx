import { useNavigate } from 'react-router-dom';
import Button from '~/components/Button';
import { useCountState } from '~/hooks/store/count';
import Todo from '~/containers/Todo';

import './App.css';

function App() {
  const navigate = useNavigate();

  const count = useCountState((state) => state.count);
  const addCount = useCountState((state) => state.addCount);

  const handleGoToReactQuery = () => {
    navigate('/rick');
  };

  return (
    <>
      <Button onClick={handleGoToReactQuery}>Fetch exemple with React Query</Button>
      <h1>Vite + React + Zustand</h1>
      <div className="card">
        <Button onClick={addCount}>count is {count}</Button>
      </div>
      <Todo />
    </>
  );
}

export default App;
