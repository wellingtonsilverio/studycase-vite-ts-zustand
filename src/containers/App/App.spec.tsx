import { render } from '@testing-library/react';
import App from '.';

describe('App', () => {
  it('should screen', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
