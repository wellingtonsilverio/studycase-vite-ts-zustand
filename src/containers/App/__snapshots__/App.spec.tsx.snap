// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render screen 1`] = `
<div>
  <h1>
    Vite + React + Zustand
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
  </div>
  <div
    class="Todo"
  >
    <h1>
      Lista de afazeres
    </h1>
    <div
      class="Input"
    >
      <input
        id="title"
      />
      <button>
        Adicionar
      </button>
    </div>
    <ul />
  </div>
</div>
`;
